### DISCUSSION TITLE
How to pass IXamlMetadataProvider to an XamlIsLand?

### DISCUSSION THREAD

**Discussion by oldoldman** [microsoft/WindowsAppSDK#3615] [General]
Hi guys,
I an UWP app, I can implement IXamlMetadataProvider just as follow
```c++
struct App : ApplicationT<App,IXamlMetadataProvider> {...}
```
when run, an instance of App will be passed to the Xaml runtime, which will query the IXamlMetadataProvider from the App instance. 
In case of Win32 app that hosts Xaml UI, I have no idea of how to pass the IXamlMetadataProvider to Xaml framework. I have read [this](https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/host-custom-control-with-xaml-islands-cpp) document. It created a global instance of hostApp (which inplements IXamlMetadataProvider interface by derive from XamlApplication), but I found no interaction between hostApp and Xaml runtime. Any suggestion? Many thanks!

**Comment by lhak** [comment]
Maybe check this repository: https://github.com/CommunityToolkit/Microsoft.Toolkit.Win32 

It contains the implementation of XAML islands for .net core 3.1.
